persistence/sql/sql_process_store.go

```diff
@@ -62,20 +63,92 @@ func (p sqlProcessStoreImpl) doStartProcessTx(
 	prcExeId := uuid.MustNewUUID()
 	hasNewWorkerTask := false
 
-	err := tx.InsertCurrentProcessExecution(ctx, extensions.CurrentProcessExecutionRow{
+	if req.ProcessStartConfig == nil {
```

Comment:
I don't feel like it's necessary to require this field. All the configs inside are "optional".

Reply:
default to reject_duplicate?

Reply:
In cadence the default is “failed only”(if previous exit abnormally). Then Temporal changed it to “allow if not running “. The reason behind that is that new users would just ask why their workflow cannot be started. It seems more friendly to new users. I would trust the decision from Temporal team for this default behavior